php/src/vaas/Vaas.php

```diff
@@ -497,14 +492,41 @@ public function setUploadTimeout(int $UploadTimeoutInSeconds): self
      * @throws GuzzleException
      * @throws UploadFailedException
      */
-    private function UploadStream(Stream $stream, string $url, string $uploadToken)
+    private function UploadStream(ReadableStreamInterface $fileStream, string $url, string $uploadToken, int $fileSize)
     {
-        $response = $this->_httpClient->put($url, [
-            'body' => $stream,
-            'content-length' => $stream->getSize(),
-            'timeout' => $this->_uploadTimeoutInSeconds,
-            'headers' => ["Authorization" => $uploadToken]
-        ]);
+        $startTime = time();
+        $lastTimestamp = time();
+        $fileStream->on('data', function () use(&$lastTimestamp, $startTime) {
+            $timeElapsedSinceLastTimestamp = time() - $lastTimestamp;
+            $timeElapsed = time() - $startTime;
+            if ($timeElapsedSinceLastTimestamp >= 5) {
+                $lastTimestamp = time();
+                $this->log("debug", "elapsed time: " . $timeElapsed);
+                $websocket = $this->_vaasConnection->GetAuthenticatedWebsocket();
+                $websocket->ping();
+            }
+        });
+
+        $timeoutTimer = LOOP::addTimer($this->_uploadTimeoutInSeconds, function () {
```

Comment:
Maybe use

$browser = $browser->withTimeout(10.0);

https://reactphp.org/http/#timeouts

Reply:
yeah, I had the problem, that it did not call the loop when having nothing in it. so it behaved differently. but I will give it a new try, since I tested it before using the await() function

Reply:
Ok. Seens to work with the withTimout now